strongbox-storage/strongbox-storage-core/src/main/java/org/carlspring/strongbox/storage/repository/Repository.java

```diff
@@ -35,6 +35,8 @@
 
     boolean allowsRedeployment();
 
+    boolean allowsUnpublish();
```

Comment:
I have the feeling that it would be a better place to place this in the `NpmRepositoryFeatures` class instead, as, (to me), this feels like a layout-specific feature. I mean... if you can delete the artifact, it generally means that you can unpublish it, right? (I'm just thinking out loud here...).

@strongbox/oss-developers , @LeifBrooks , @pyashwanthkumar : What do you guys think...?

Reply:
Since `unpublish` is specific to `Npm`, it would be nice to keep it in a class related to `Npm` such as `NpmRepositoryFeatures`. This would improve the cohesiveness of the class `NpmRepositoryFeatures` and at the same time, it's loosely coupled.